system_status/system_status.py

```diff
@@ -0,0 +1,104 @@
+import boto3
+import json
+import logging
+import os
+import sqlalchemy
+
+from notifications_utils.system_status import determine_notification_status, determine_site_status, TEMPLATES, THRESHOLDS
+
+URL_API = os.getenv("system_status_api_url")
+URL_ADMIN = os.getenv("system_status_admin_url")
+DB_CONN_STRING = os.getenv("sqlalchemy_database_reader_uri")
+BUCKET_NAME = os.getenv("system_status_bucket_name")
+
+# get list of UUIDS from TEMPLATES
+TEMPLATE_IDS = ",".join(["'{}'".format(x) for x in [template_id for notification_type in TEMPLATES for template_id in TEMPLATES[notification_type].values()]])
+
+
+# Build SQL query
+SQL = ("SELECT      template_id,  AVG(extract(epoch from (n.sent_at - n.created_at)))*1000  as time_to_send "
+        "FROM       notifications n "
+        "WHERE      JOB_ID is null "
+        "AND        template_id in ({}) "
+        "AND        created_at >= NOW() - INTERVAL '5 MINUTES' "
+        "GROUP BY   template_id ".format(TEMPLATE_IDS))
+
+def lambda_handler(event, context):
+    
+    try:
+        # connect to postgres db
+        db = sqlalchemy.create_engine(DB_CONN_STRING, future=True)
+        with db.connect() as conn:
+            dbresults = conn.execute(sqlalchemy.sql.text(SQL))
+            conn.commit()    
+
+    except sqlalchemy.exc.OperationalError as e:
+        logging.error("System-status lambda (db connection): {}".format(e))
+        db.dispose();
+
+        return {
+            'statusCode': 500
+        }
+    except Exception as e:
+        logging.error("System-status lambda (db connection): unknown error: {}".format(e))
+        db.dispose();
+
+        return {
+            'statusCode': 500
+        }
+    
+    try:
+        (email_status, sms_status) = determine_notification_status(dbresults)
+    except Exception as e:
+        logging.error("System-status lambda error (determine_notification_status): {}".format(e))
+        return {
+            'statusCode': 500
+        }
+
+    
+    db.dispose()
+
+    # check api and admin response times
+    api_status = determine_site_status(URL_API, THRESHOLDS['api'])
+    admin_status = determine_site_status(URL_ADMIN, THRESHOLDS['admin'])
```

Comment:
should we have error handling on the above two? aka there was no api_status data or admin_status data, but we have the rest? In that case - information for those fields would be empty but not the rest of the json response object?

Reply:
So I added error handling inside of `determine_site_status` which basically marks them as down if an unhandled exception occurs.  Maybe that is a bit confusing though since you can't see it here?

Reply:
o i followed see what you did. this works! thanks!